ledger-core/v2/network/consensus/serialization/globula_protocol_test.go

```diff
@@ -253,172 +216,173 @@ func TestGlobulaConsensusPacketBody_Phases(t *testing.T) {
-	phase1p.EncryptableBody.(*GlobulaConsensusPacketBody).PulsarPacket.setData(bs)
-
-	tests := []struct {
-		name       string
-		packetType phases.PacketType
-		size       int64
-		packet     Packet
-	}{
-		{
-			"phase0",
-			phases.PacketPhase0,
-			401,
-			phase1p,
-		},
-		{
-			"phase1",
-			phases.PacketPhase1,
-			404,
-			phase1p,
-		},
-		{
-			"phase2",
-			phases.PacketPhase2,
-			90,
-			p,
-		},
-		{
-			"phase3",
-			phases.PacketPhase3,
-			219,
-			p,
-		},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			p := test.packet
-
-			p.Header.setPacketType(test.packetType)
-			p.Header.SetFlag(0)
-
-			s, err := p.SerializeTo(context.Background(), buf, digester, signer)
-			require.NoError(t, err)
-			require.EqualValues(t, test.size, s)
-			require.EqualValues(t, s, p.Header.getPayloadLength())
-
-			require.NotEmpty(t, p.PacketSignature)
-
-			p2 := Packet{}
-
-			_, err = p2.DeserializeFrom(context.Background(), buf)
-			require.NoError(t, err)
-		})
-	}
-}
-
-func TestGlobulaConsensusPacketBody_Phases_Flag0Reset(t *testing.T) {
-	data := pulse.NewPulsarData(100000, 10, 10, longbits.Bits256{})
-
-	pu := adapters.NewPulse(data)
-	ph, err := host.NewHost("127.0.0.1:1")
-	require.NoError(t, err)
-	th, err := net.ResolveTCPAddr("tcp", "127.0.0.1:2")
-	require.NoError(t, err)
-	pp := pulsenetwork.NewPulsePacketWithTrace(context.Background(), &pu, ph, th, 0)
-
-	bs, err := packet.SerializePacket(pp)
-	require.NoError(t, err)
-
-	buf := bytes.NewBuffer(make([]byte, 0, packetMaxSize))
-
-	p := Packet{
-		Header: Header{
-			SourceID:   123,
-			TargetID:   456,
-			ReceiverID: 789,
-		},
-		EncryptableBody: &GlobulaConsensusPacketBody{},
-	}
-	p.Header.setProtocolType(ProtocolTypeGlobulaConsensus)
-
-	phase1p := p
-	phase1p.EncryptableBody = &GlobulaConsensusPacketBody{}
-	phase1p.EncryptableBody.(*GlobulaConsensusPacketBody).PulsarPacket.setData(bs)
-
-	p.Header.SetFlag(0)
-	phase1p.Header.SetFlag(0)
-
-	tests := []struct {
-		name       string
-		packetType phases.PacketType
-		size       int
-		packet     Packet
-	}{
-		{
-			"phase0",
-			phases.PacketPhase0,
-			88,
-			phase1p,
-		},
-		{
-			"phase1",
-			phases.PacketPhase1,
-			91,
-			phase1p,
-		},
-		{
-			"phase2",
-			phases.PacketPhase2,
-			90,
-			p,
-		},
-		{
-			"phase3",
-			phases.PacketPhase3,
-			219,
-			p,
-		},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			p := test.packet
-
-			p.Header.setPacketType(test.packetType)
-			p.Header.ClearFlag(0)
-
-			s, err := p.SerializeTo(context.Background(), buf, digester, signer)
-			require.NoError(t, err)
-			require.EqualValues(t, test.size, s)
-			require.EqualValues(t, s, p.Header.getPayloadLength())
-
-			require.NotEmpty(t, p.PacketSignature)
-
-			p2 := Packet{}
-
-			_, err = p2.DeserializeFrom(context.Background(), buf)
-			require.NoError(t, err)
-		})
-	}
-}
+//
+// func TestGlobulaConsensusPacketBody_Phases_Flag0(t *testing.T) {
```

Comment:
@cyraxred here I need your help, I'm not quite sure what actually is going on here.

old NewPulsePacketWithTrace is protobuf packet, as far as I understand, how and why it goes to .setData. 

It looks like theese tests are no longer valid at all and could be deleted

Reply:
@AndreyBronin maybe you know something about it?

Reply:
ok, fixed the tests with proper pulsar_packet